test/http-test.js

```diff
@@ -446,6 +446,26 @@ describe('HTTP', function() {
     }
   });
 
+  it('should create a transaction using the change address', async () => {
+    const receive = await wallet.createChange('default');
+    const change1 = await wallet.createChange('default');
+    const change2 = await wallet.createChange('default');
+    const change3 = await wallet.createChange('default');
+
+    const tx = await wallet.createTX({
+      sign: false,
+      outputs: [{
+        address: receive.address,
+        value: 50000
+      }],
+      change: change2.address
+    });
+
+    assert(!tx.outputs.some(o => o.address === change1.address));
+    assert(tx.outputs.some(o => o.address === change2.address));
+    assert(!tx.outputs.some(o => o.address === change3.address));
+  });
+
```

Comment:
Ok here's an alternative that really specifies what the test is trying to do, as acutely as possible. Some things to note:
- Don't let the wallet pick the address, let's hard-code it since the whole point is we are passing the address as an option.
- No need for testing false-negatives here, really. Either the transaction is exactly what we expect or it isn't.
- By setting fee rate to 0 and sorting to false, I know the tx will have the exact form I want (change address last, all values totally predictable without any fees -- nice to have regtest!)




```js
  it('should create a transaction using the change address', async () => {
    const receive =
      Address.fromPubkeyhash(Buffer.alloc(20, 1)).toString('regtest');
    const change =
      Address.fromPubkeyhash(Buffer.alloc(20, 2)).toString('regtest');

    const tx = await wallet.createTX({
      rate: 0,
      sort: false,
      outputs: [{
        address: receive,
        value: 50000
      }],
      change: change
    });

    let total = 0;
    for (const input of tx.inputs)
      total += input.coin.value;

    assert.strictEqual(tx.outputs.length, 2);

    assert.strictEqual(tx.outputs[0].address, receive);
    assert.strictEqual(tx.outputs[1].address, change);
    assert.strictEqual(tx.outputs[0].value, 50000);
    assert.strictEqual(tx.outputs[1].value, total - 50000);
  });
```

Reply:
Valid points,  refactored.